app/modules/topics/components/Editor/index.js

```diff
@@ -47,41 +69,78 @@ class PureEditor extends React.Component<Props> {
     onSave();
   };
 
+  componentWillUnmount = (): void => {
+    const { topic, discard } = this.props;
+
+    if (topic.isDirty) discard(topic.id);
+  };
+
   fetchCondition = (topic: ?m.Topic): boolean => {
     return (topic == null || !topic.isContentFetched);
   };
 
   renderEditor = (topic: m.Topic): React.Node => {
-    const { t } = this.props;
+    const { t, setDirty, beforeUnloadHandler } = this.props;
+
+    // Prompt when user refreshes window with unsaved changes
+    if (topic.isDirty) {
+      window.addEventListener('beforeunload', beforeUnloadHandler);
+    }
+    else {
+      window.removeEventListener('beforeunload', beforeUnloadHandler);
+    }
```

Comment:
- Don't add event listeners in `render`; this code is run every time the component is re-rendered.
- Since this code is currently not doing anything, it's probably best to remove it entirely to prevent confusion?